src/Simplify.elm

```diff
@@ -4913,8 +4902,13 @@ resultWithDefaultChecks checkInfo =
 
 
 collectionFilterChecks : Collection -> CheckInfo -> List (Error {})
-collectionFilterChecks collection ({ parentRange, fnRange, firstArg, secondArg } as checkInfo) =
-    case Maybe.andThen (collection.determineSize checkInfo.lookupTable) checkInfo.secondArg of
+collectionFilterChecks collection checkInfo =
+    let
+        collectionArg : Maybe (Node Expression)
+        collectionArg =
+            secondArg checkInfo
+    in
```

Comment:
I'm not fully convinced since you usually only access second, third once in the ...checks functions and not repeatedly.
I agree tho that converting to an `Array` is not worth the cost

Reply:
Right. Let's try this out and see how we like it in practice